compiler/lambdalifting.nim

```diff
@@ -516,6 +525,12 @@ proc detectCapturedVars(n: PNode; owner: PSym; c: var DetectionPass) =
       detectCapturedVars(n[namePos], owner, c)
   of nkReturnStmt:
     detectCapturedVars(n[0], owner, c)
+  of nkIdentDefs:
+    if not containsCallKinds(n[1]):
+      detectCapturedVars(n[^1], owner, c)
```

Comment:
Why not always simply do `detectCapturedVars(n[^1], owner, c)`?

Reply:
it will fails on `zero-functional`'s   `gensym4090: seq[type(_ZF_iterType()[1])]`

Reply:
How so? `n[^1]` is the `value` part of `var v: T = value` and it is empty for your snippet.

Reply:
so that it avoid `detectCapturedVars` of `_ZF_iterType` then it loss env

Reply:
You misunderstand me. I'm asking why the patch is not just:

```suggestion
  of nkIdentDefs:
    detectCapturedVars(n[^1], owner, c)
```

Without the `if`.

Reply:
I tried at first time, it looks much cleaner and will solve https://github.com/nim-lang/Nim/issues/21737, but `detectCapturedVars` also calls `addClosureParam`, if we avoid processing `detectCapturedVars` on `gensym4090: seq[type(_ZF_iterType()[1])]` (the type modifier), we will loss env param for `_ZF_iterType`

Reply:
Makes no sense to me. So we need some special case code for the `typ` position of the variable declaration then but that's it.

Reply:
I tried to add extra code handle all cases, but no luck, https://github.com/nim-lang/Nim/issues/21737 result to `cgen` error, others passed.